### ISSUE TITLE
grains.append convert argument incorrectly converts null/None to list items

### ISSUE THREAD

**Issue by oliver-dungey** [saltstack/salt#35094]
### Description of Issue/Question
I have a simple roles system using grains. It's not your standard list of roles but a series of roles that can optionally have a list of specific environments/systems that the role applies to - the grains data looks like this:
```
roles:
  web_server:
     - productionA
     - systemTestB
  database_server:
  some_other_thing:
```
To allow me to modify or add new roles (with or without the environment/systems) I have found that **grains.append** works beautifully with its recent support for nested lists: so long as I supply the necessary **convert=True** argument. The only fly in the ointment is that nulls/empty-lists get converted to a list item too, and there seems no point/purpose/logic to having a list item of null/None (unless someone explicitly asks for that). Therefore running this command (in the context of the above grain data):

`salt 'minion' grains.append 'roles:database_server [envA, envB, envC] True`

Results in this:

```
local:
    ----------
    roles:
        ----------
        web_server:
         - productionA
         - systemTestB
        database_server:
        - None
        - envA
        - envB
        - envC
        some_other thing:
```
There seem to be quite a few different conditions where you can end up with **None** list items although I'm not yet sure if this issue is only present in **append**. For my scenario its not a big problem as I'm only checking for the existence of roles but I can imagine this would be a horrible pain for other situations.

### Setup
CentOS 6.7
salt-call 2016.3.1 (Boron)

### Versions Report
```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Jul 23 2015, 15:22:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 6.7 Final
        machine: x86_64
        release: 2.6.32-573.el6.x86_64
         system: Linux
        version: CentOS 6.7 Final
```

**Comment by Ch3LL** [comment]
@oliver-dungey looks like I am able to replicate this in 2016.3.1:

```
 ch3ll@ch3ll-master  ~  sudo salt '10*'  grains.append 'roles:database_server' '[envA, envB, envC]' convert=True 

10.1.45.255:
    ----------
    roles:
        ----------
        database_server:
            - None
            - envA
            - envB
            - envC
        some_other_thing:
            None
        web_server:
            - productionA
            - systemTestB
```

Looks we definitely need to clean this up. Thanks for the heads up.

**Comment by rallytime** [comment]
@oliver-dungey This is fixed by #35407

**Comment by oliver-dungey** [comment]
@rallytime Excellent, thanks.
